src/Repositories/CustomUserRepository.php

```diff
@@ -1,49 +1,37 @@
 <?php
 
-namespace Salt\Auth0\Repositories;
+declare(strict_types=1);
 
-use Auth0\Login\Auth0JWTUser;
-use Auth0\Login\Auth0User;
-use Auth0\Login\Repository\Auth0UserRepository;
+namespace Salt\Auth0\Repositories;
 
 use Salt\Auth0\Models\User;
 use Salt\Auth0\Requesters\Auth0ApiRequester;
+use Auth0\Laravel\Contract\Auth\User\Repository;
 
-class CustomUserRepository extends Auth0UserRepository
+class CustomUserRepository implements Repository
 {
     /**
-     * Get an existing user or create a new one
+     * Updates or creates a user from stateful session data
+     * provided by authentication with auth0.
      *
-     * @param array $profile - Auth0 profile
+     * The user returned in this function must implement the interface
+     * Auth0\Laravel\Contract\Model\Stateful\User
      *
-     * @return User
-     */
-    protected function upsertUser($profile)
-    {
-        $user = User::firstOrCreate(['email' => $profile['email']], [
-            'sub' => $profile['sub'] ?? '',
-            'name' => $profile['name'] ?? '',
-        ]);
-
-        return $user;
-    }
-
-    /**
-     * Custom method to create or update a user on sign up and store them locally
+     * @param $data The user data provided by auth0
      *
-     * @param array $data - user sign up data
+     * @return $user The stateful user
      *
-     * @return User
+     * @inheritdoc
      */
-    public function updateOrCreate($data): object
-    {
-        // Update or create user
+    public function fromSession(
+        array $data
+    ): ?\Illuminate\Contracts\Auth\Authenticatable {
         $user = User::updateOrCreate(
             [
                 'email' => $data['email'],
             ],
             [
-                'name' => $data['first_name'] . " " . $data['last_name'],
+                'name' => $data['name'],
             ]
         );
```

Comment:
Move the previous code for creating and updating users into the new `fromSession` method.